### PR TITLE
What's New: SignalR update per feedback

### PR DESCRIPTION
**Pull request by wadepickett**
[Internal review preview: What's New: SignalR section](https://review.learn.microsoft.com/en-us/aspnet/core/release-notes/aspnetcore-9.0?view=aspnetcore-9.0&branch=pr-en-us-33341#signalr-supports-trimming-and-native-aot)

Fixes https://github.com/dotnet/AspNetCore.Docs/issues/33054
Contributes to https://github.com/dotnet/AspNetCore.Docs/issues/33031

Adding in suggestions provided after merged of #33318:

- Revert limitations back to https://github.com/dotnet/release-notes-drafts/edit/main/9.0/preview7/aspnetcore.md#limitations: (per eerhardt)
  - Replace mention of "client proxy generation" with "strongly typed hubs". (per Brennan)
  - Keep issue link update in limitations added by guardrex.

- Remove file SignalRChatAOTExample.http since the code doesn't have a todos endpoint.

### CODE REVIEW

File: aspnetcore/release-notes/aspnetcore-9/includes/signalr-trimming-aot.md [hunk 1]
```diff
@@ -20,7 +20,10 @@ The preceding example produces a native Windows executable of 10 MB and a Linux
...
+* Only the JSON protocol is currently supported:
+  * As shown in the preceding code, apps that use JSON serialization and Native AOT must use the `System.Text.Json` Source Generator. 
+  * This follows the same approach as minimal APIs.
+* On the SignalR server, Hub method parameters of type `IAsyncEnumerable<T>` and `ChannelReader<T>` where `T` is a ValueType (i.e. `struct`) aren't supported. Using these types results in a runtime exception at startup in development and in the published app. For more information, see [SignalR: Using IAsyncEnumerable&lt;T&gt; and ChannelReader&lt;T&gt; with ValueTypes in native AOT (`dotnet/aspnetcore` #56179)](https://github.com/dotnet/aspnetcore/issues/56179).
+* [Strongly typed hubs](xref:signalr/hubs#strongly-typed-hubs) aren't supported with Native AOT (`PublishAot`). Using strongly typed hubs with Native AOT will result in warnings during build and publish, and a runtime exception. Using a Using strongly typed hubs with trimming (`PublishedTrimmed`) is supported.
```
Comment (BrennanConroy) [inline aspnetcore/release-notes/aspnetcore-9/includes/signalr-trimming-aot.md:27]:
```suggestion
* [Strongly typed hubs](xref:signalr/hubs#strongly-typed-hubs) aren't supported with Native AOT (`PublishAot`). Using strongly typed hubs with Native AOT will result in warnings during build and publish, and a runtime exception. Using strongly typed hubs with trimming (`PublishedTrimmed`) is supported.
```

**Review by eerhardt — APPROVED** [review]
Thanks!

File: aspnetcore/release-notes/aspnetcore-9/includes/signalr-trimming-aot.md [hunk 2]
```diff
@@ -20,7 +20,10 @@ The preceding example produces a native Windows executable of 10 MB and a Linux
...
-* On a SignalR server, Hub method parameters of type `IAsyncEnumerable<T>` and `ChannelReader<T>` where `T` is a ValueType (i.e. `struct`) aren't supported. These will cause runtime exceptions at startup during development and in the published app. For more information, see [SignalR: Using IAsyncEnumerable&lt;T&gt; and ChannelReader&lt;T&gt; with ValueTypes in native AOT (`dotnet/aspnetcore` #56179)](https://github.com/dotnet/aspnetcore/issues/56179).
-* Client proxy generation is not supported in PublishAot. Client proxy generation will result in warnings at build-time and works with `PublishTrimmed` with no warnings.
-* Custom awaitables on the server are not supported in trimming or AOT. The feature switch `Microsoft.AspNetCore.SignalR.Hub.IsCustomAwaitableSupported` AppContext switch and `SignalRCustomAwaitableSupport` MSBuild property, is disabled by default in trimmed and AOT compiled apps.
+* Only the JSON protocol is currently supported:
+  * As shown in the preceding code, apps that use JSON serialization and Native AOT must use the `System.Text.Json` Source Generator.
```
Comment (BrennanConroy) [inline aspnetcore/release-notes/aspnetcore-9/includes/signalr-trimming-aot.md:24]: > As shown in the preceding code

Where is the preceding code?

Comment (BrennanConroy) [reply aspnetcore/release-notes/aspnetcore-9/includes/signalr-trimming-aot.md:24]: Oh, it's the code link above, nvm 👍

**Review by BrennanConroy — APPROVED** [review]

File: aspnetcore/release-notes/aspnetcore-9/includes/signalr-trimming-aot.md (hunk 2 above)
Comment (wadepickett) [reply aspnetcore/release-notes/aspnetcore-9/includes/signalr-trimming-aot.md:24]: [Preview for that section is here](https://review.learn.microsoft.com/en-us/aspnet/core/release-notes/aspnetcore-9.0?view=aspnetcore-9.0&branch=pr-en-us-33341#signalr-supports-trimming-and-native-aot)

**Comment by wadepickett** [comment]
My apologies, I didn't realize the source we were pulling/syncing from the release notes repo until Dan mentioned it to me just after closing the release notes for ship.  All I knew was that I was to pull from the discussion in #33031 and thought the source was completely here.  Next time I will watch the release-notes repo to make sure I have any updates from there synced right away in the What's New here in docs.

**Pull request merged**